Unable to run make check-change

As part of the v0.81.0/v1.0.0-rcv0014 release, I ran the `make check-change` target but ran into multiple issues. Here is a copy of the output on my terminal:

```sh
❯ make check-changes PREVIOUS_VERSION=v1.0.0-rcv0013 MODSET=stable
make: Nothing to be done for 'check-changes'.

❯ make -B check-changes PREVIOUS_VERSION=v1.0.0-rcv0013 MODSET=stable
mkdir -p /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools
cd /home/pablo.baeyens/Source/otel/opentelemetry-collector/internal/tools && go build -o /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools/yq -trimpath github.com/mikefarah/yq/v4

❯ make -B check-changes PREVIOUS_VERSION=v1.0.0-rcv0013 
mkdir -p /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools
cd /home/pablo.baeyens/Source/otel/opentelemetry-collector/internal/tools && go build -o /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools/yq -trimpath github.com/mikefarah/yq/v4

❯ make -B check-changes MODSET=stable
mkdir -p /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools
cd /home/pablo.baeyens/Source/otel/opentelemetry-collector/internal/tools && go build -o /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools/yq -trimpath github.com/mikefarah/yq/v4
PREVIOUS_VERSION not defined
usage: make check-changes PREVIOUS_VERSION=<version eg 0.52.0> MODSET=beta
exit 1
make: *** [Makefile:444: check-changes] Error 1
```

To summarize the issues found:

1. The target is not identified as 'phony', so I have to force running it with `-B`
2. The target does not seem to do anything if I pass the (pressumably) correct inputs
3. The target does not fail if I omit `MODSET`, but does if I omit `PREVIOUS_VERSION`

If I remove the `ifndef` checks I can get some output (using `NORMALIZED_PREVIOUS_VERSION` instead of `PREVIOUS_VERSION` instead):

```
make -B check-changes NORMALIZED_PREVIOUS_VERSION=v1.0.0-rcv0013 MODSET=stable
mkdir -p /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools
cd /home/pablo.baeyens/Source/otel/opentelemetry-collector/internal/tools && go build -o /home/pablo.baeyens/Source/otel/opentelemetry-collector/.tools/yq -trimpath github.com/mikefarah/yq/v4
No stable modules have changed since v1.0.0-rcv0013
No need to release stable.
make: *** [Makefile:436: check-changes] Error 1
```

I am using GNU make 4.3:

```
❯ make --version
GNU Make 4.3
Built for x86_64-pc-linux-gnu
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

--- a/Makefile
+++ b/Makefile
@@ -432,44 +432,8 @@ push-tags: $(MULTIMOD)
 	done;
 
 .PHONY: check-changes
-check-changes: $(YQ)
-ifndef MODSET
-	@echo "MODSET not defined"
-	@echo "usage: make check-changes PREVIOUS_VERSION=<version eg 0.52.0> MODSET=beta"
-	exit 1
-endif
-ifndef PREVIOUS_VERSION
-	@echo "PREVIOUS_VERSION not defined"
-	@echo "usage: make check-changes PREVIOUS_VERSION=<version eg 0.52.0> MODSET=beta"
-	exit 1
-else
-ifeq (, $(findstring v,$(PREVIOUS_VERSION)))
-NORMALIZED_PREVIOUS_VERSION="v$(PREVIOUS_VERSION)"
-else
-NORMALIZED_PREVIOUS_VERSION="$(PREVIOUS_VERSION)"
-endif
-endif
-	@all_submods=$$($(YQ) e '.module-sets.*.modules[] | select(. != "go.opentelemetry.io/collector")' versions.yaml | sed 's/^go\.opentelemetry\.io\/collector\///'); \
-	mods=$$($(YQ) e '.module-sets.$(MODSET).modules[]' versions.yaml | sed 's/^go\.opentelemetry\.io\/collector\///'); \
-	changed_files=""; \
-	for mod in $${mods}; do \
-		if [ "$${mod}" == "go.opentelemetry.io/collector" ]; then \
-			changed_files+=$$(git diff --name-only $(NORMALIZED_PREVIOUS_VERSION) -- $$(printf '%s\n' $${all_submods[@]} | sed 's/^/:!/' | paste -sd' ' -) | grep -E '.+\.go$$'); \
-		elif ! git rev-parse --quiet --verify $${mod}/$(NORMALIZED_PREVIOUS_VERSION) >/dev/null; then \
-			echo "Module $${mod} does not have a $(NORMALIZED_PREVIOUS_VERSION) tag"; \
-			echo "$(MODSET) release is required."; \
-			exit 0; \
-		else \
-			changed_files+=$$(git diff --name-only $${mod}/$(NORMALIZED_PREVIOUS_VERSION) -- $${mod} | grep -E '.+\.go$$'); \
-		fi; \
-	done; \
-	if [ -n "$${changed_files}" ]; then \
-		echo "The following files changed in $(MODSET) modules since $(NORMALIZED_PREVIOUS_VERSION): $${changed_files}"; \
-	else \
-		echo "No $(MODSET) modules have changed since $(NORMALIZED_PREVIOUS_VERSION)"; \
-		echo "No need to release $(MODSET)."; \
-		exit 1; \
-	fi
+check-changes: $(MULTIMOD)
+	$(MULTIMOD) diff -p $(PREVIOUS_VERSION) -m $(MODSET)
 
 .PHONY: prepare-release
 prepare-release:

--- a/internal/tools/go.mod
+++ b/internal/tools/go.mod
@@ -13,7 +13,7 @@ require (
 	github.com/pavius/impi v0.0.3
 	go.opentelemetry.io/build-tools/checkdoc v0.9.0
 	go.opentelemetry.io/build-tools/chloggen v0.9.0
-	go.opentelemetry.io/build-tools/crosslink v0.8.0
+	go.opentelemetry.io/build-tools/crosslink v0.9.0
 	go.opentelemetry.io/build-tools/multimod v0.9.0
 	go.opentelemetry.io/build-tools/semconvgen v0.9.0
 	golang.org/x/exp v0.0.0-20230522175609-2e198f4a06a1

